pyogrio/core.py

```diff
@@ -1,5 +1,8 @@
 """Core functions to interact with OGR data sources."""
 
+from pathlib import Path
+from typing import Optional, Union
```

Comment:
I don't believe we've integrated typing elsewhere?  Is the idea here to start progressively adding it?

Reply:
Actually I don't recall giving it a conscious thought... so I suppose it was rather out of habit as I normally write typed. I think it might be a good idea to start using it, espacially for pyogrio I think it wouldn't be to hard for most parameters to type them. But I wouldn't mind removing the typing either.